pyopia/background.py

```diff
@@ -151,6 +151,44 @@ def correct_im_fast(imbg, imraw):
     return im_corrected
 
 
+def correct_im_accurate_divide(imbg, imraw):
+    '''
+    Corrects raw image by dividing the background and scaling the output
+    (see Davies et al., 2017: https://doi.org/10.1016/j.marpolbul.2016.11.063)
+
+    There is a small chance of clipping of imc in both crushed blacks and blown
+    highlights if the background or raw images are very poorly obtained
+
+    Parameters
+    -----------
+    imbg : float64
+        background averaged image
+    imraw : float64
+        raw image
+
+    Returns
+    -------
+    im_corrected : float64
+        corrected image, same type as input
+    '''
+
+    image_shape = np.shape(imbg)
+    if len(image_shape) > 2:
+        # change the zero_value pixels in each channel to prevent RuntimeWarning: 'divide by zero'
+        for i in range(imbg.shape[2]):
+            imbg[:, :, i][imbg[:, :, i] == 0] = 1 / 255
+        else:
+            imbg[imbg == 0] = 1 / 255
```

Comment:
I think you can replace all the preceding lines with this: 

imbg = np.clip(imbg, a_min=1/255)